### PR TITLE
feat: add `add_handler_after`, `add_handler_before`, `replace_handler`

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

feat: add `add_handler_after`, `add_handler_before`, `replace_handler` for `HeartbeatHandlerGroupBuilder`

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: src/meta-srv/src/handler.rs [hunk 1]
```diff
@@ -489,50 +487,128 @@ impl HeartbeatHandlerGroupBuilder {
...
+                break;
+            }
+            cursor.move_next();
+        }
+
+        ensure!(found, error::HandlerNotFoundSnafu { name });
+        Ok(())
+    }
+
+    fn add_handler(&mut self, handler: impl HeartbeatHandler + 'static) {
```
Comment (fengjiachun) [inline src/meta-srv/src/handler.rs:603]: Better to rename to `add_handler_last`

File: src/meta-srv/src/error.rs [hunk 2]
```diff
@@ -733,6 +733,13 @@ pub enum Error {
         #[snafu(implicit)]
         location: Location,
     },
+
+    #[snafu(display("Handler: {} is not found", name))]
```
Comment (fengjiachun) [inline src/meta-srv/src/error.rs:737]:
```suggestion
    #[snafu(display("Handler not found: {}", name))]
```

File: src/meta-srv/src/handler.rs [hunk 3]
```diff
@@ -489,50 +487,128 @@ impl HeartbeatHandlerGroupBuilder {
...
+            handlers: self.handlers.into_iter().collect(),
+            pushers: self.pushers,
+        }
+    }
+
+    /// Adds the handler after the specified handler.
+    pub fn add_handler_after(
+        &mut self,
+        handler: impl HeartbeatHandler + 'static,
+        after: &'static str,
```
Comment (fengjiachun) [inline src/meta-srv/src/handler.rs:541]:
```suggestion
        name: &'static str,
```

Let's place the name argument before the handler.

File: src/meta-srv/src/handler.rs [hunk 4]
```diff
@@ -489,50 +487,128 @@ impl HeartbeatHandlerGroupBuilder {
...
+            }
+            cursor.move_next();
+        }
+
+        ensure!(found, error::HandlerNotFoundSnafu { name: after });
+        Ok(())
+    }
+
+    /// Adds the handler before the specified handler.
+    pub fn add_handler_before(
```
Comment (fengjiachun) [inline src/meta-srv/src/handler.rs:560]: Let's place the name argument before the handler.

File: src/meta-srv/src/handler.rs [hunk 5]
```diff
@@ -489,50 +487,128 @@ impl HeartbeatHandlerGroupBuilder {
...
+        }
+        ensure!(found, error::HandlerNotFoundSnafu { name: before });
+        Ok(())
+    }
+
+    /// Replaces the handler with the specified name.
+    pub fn replace_handler(
+        &mut self,
+        handler: impl HeartbeatHandler + 'static,
+        name: &'static str,
```
Comment (fengjiachun) [inline src/meta-srv/src/handler.rs:584]: Let's place the name argument before the handler.

**Review by fengjiachun — APPROVED** [review]
LGTM

File: src/meta-srv/src/handler.rs [hunk 6]
```diff
@@ -458,25 +447,34 @@ pub struct HeartbeatHandlerGroupBuilder {
...
+    region_lease_handler: Option<RegionLeaseHandler>,
 
     /// The plugins.
     plugins: Option<Plugins>,
 
     /// The heartbeat response pushers.
     pushers: Pushers,
+
+    /// The group of heartbeat handlers.
+    handlers: LinkedList<NameCachedHandler>,
```
Comment (evenyag) [inline src/meta-srv/src/handler.rs:459]: I think Vec is enough as we won't have too many handlers.

Comment (WenyXu) [reply src/meta-srv/src/handler.rs]: Resoved in [bcfc11d](https://github.com/GreptimeTeam/greptimedb/pull/4788/commits/bcfc11d465a7212bc4e479ce3fd77e868b77baa7)

File: src/meta-srv/src/lib.rs [hunk 7]
```diff
@@ -17,6 +17,7 @@
 #![feature(assert_matches)]
 #![feature(extract_if)]
 #![feature(option_take_if)]
+#![feature(linked_list_cursors)]
```
Comment (evenyag) [inline src/meta-srv/src/lib.rs:20]:
```suggestion
```

**Review by evenyag — APPROVED** [review]

**Pull request merged**